src/meta_schedule/database/database.cc

```diff
@@ -59,12 +60,8 @@ Workload Workload::FromJSON(const ObjectRef& json_obj) {
       String b64_mod = Downcast<String>(json_array->at(1));
       std::string json_mod = Base64Decode(b64_mod);
       mod = Downcast<IRModule>(LoadJSON(json_mod));
+      std::stringstream(str_shash) >> shash;
     }
-    // Verify SHash(mod) == shash
-    shash = tvm::StructuralHash()(mod);
-    String recalc_shash = SHash2Str(shash);
-    CHECK_EQ(recalc_shash, str_shash) << "ValueError: Structural hash changed. Given: " << str_shash
-                                      << "; Recalculated: " << recalc_shash;
```

Comment:
Since ModuleEquality information is not written to json (Workload doesn't have ModuleEquality instance), we cannot determine the appropriate hashing method here. So I had to remove this verification.

Reply:
Perhaps we could leave a TODO here in case of future changes

Reply:
To keep things clean, I want to minimize the number of unique `ModuleEquality` instances scattered throughout MS infra. Currently, only task extraction and `Database` "own" a `ModuleEquality` instance.

Given this goal, I think having `Database` compute a hash of `Workload` using its `ModuleEquality` is more natural than passing `mod_eq_name` to `Workload` and let it compute hash. Moreover, all `Workload` would end up having the same `mod_eq_name`, so it feels redundant.

Reply:
One way to restore hash verification is to do verification inside `DatabaseNode` after `Workload::FromJSON` is called, since the database does know what the `ModuleEquality` instance is supposed to be.

Reply:
Restored hash verification in `JSONDatabase`, the only place where `Workload::FromJSON` is used currently.